.github/workflows/check_interface_compatibility.yml

```diff
@@ -0,0 +1,80 @@
+name: Check interface compatibility
+
+on:
+  pull_request_target:
+
+permissions:
+  pull-requests: write
+
+jobs:
+  check-interface-compatibility:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Extract original public interfaces
+        run: |
+          make init
+          bin/public_interface.py extract > ${{ runner.temp }}/interfaces.original.json
+
+      - name: Backup original bin/public_interface.py
+        run: |
+          # Keep a copy of bin/public_interface.py to use on the base branch
+          # So we are using the same bin/public_interface.py to process old/new codebase.
+          # Because the step has credential of GitHub token to post comments,
+          # always use the original version that is in our repository.
+          cp bin/public_interface.py ${{ runner.temp }}/public_interface.py
+
+      - name: Checkout the PR
+        uses: actions/checkout@v3
+        with:
+          ref: "refs/pull/${{ github.event.number }}/merge"
+
+      - name: Extract PR public interfaces
+        run: |
+          make init
```

Comment:
Why is this needed again? Aren't all the dependencies from the base branch code, since that's the only script that's used?

Reply:
in case there is dependency change and module cannot be imported if samtranslator depends on a new pypi package.